framework/db/ActiveQuery.php

```diff
@@ -762,7 +762,7 @@ public function orOnCondition($condition, $params = [])
      */
     public function viaTable($tableName, $link, callable $callable = null)
     {
-        $modelClass = $this->primaryModel !== null ? get_class($this->primaryModel) : get_class();
+        $modelClass = $this->primaryModel !== null ? get_class($this->primaryModel) : __CLASS__;
```

Comment:
This looks really weird, using ActiveQuery and modelclass does not make any sense here. If primaryModel is null, it should fail, one can not call viaTable on a query that is not a relation...

Reply:
found the original commit that added this behavior here: https://github.com/yiisoft/yii2/commit/64d8af61a64db0b00d498800e1f20e62b4473329#diff-1776bc9373c1dbb0d6e77ba4d8081a81L762

Reply:
not related to this PR; created an issue: https://github.com/yiisoft/yii2/issues/15876